### ISSUE TITLE
No sound on clean install only with clear linux

### ISSUE THREAD

**Issue by butterflycollector** [clearlinux/distribution#1911]
Hello,

There is no sound after install of the latest 32860 clear linux and there is no sound in the live cd, either. I tried several 30xxx, 31xxx and 32xxx versions and even one from last year and there is no sound. My audio card is Creative Sound BlasterX AE-5. My motherboard is ASUS ROG Strix z390-e. My sound is working just fine in win 10, win 8.1, opensuse tumbleweed, Pop!_os and ubuntu. Last year I had tried a version of clear linux in which I had sound, but I don't want to roll back and I don't remember which one. The speakers are always plugged in the right place, no hardware changes.

currently I have the latest:

32860

`uname -a`

> Linux samara 5.6.4-937.native #1 SMP Mon Apr 13 06:03:38 PDT 2020 x86_64 GNU/Linux
> 

My alsamixer is set just fine: nothing is muted, nothing is at or close to 0; the settings are as in other distros.

The FN + +/- keyboard keys work just fine and show a decent sound level.

The sound bar at the top right corner of gnome is just fine.

gnome -> settings -> sound show that my sound card has been selected right and the sound/level/ meter bar below the selector is moving as if sound is being pushed towards the card. The "Output Device" selector says: "Line out - HDA Creative" (my PCIe Creative Soundblaster AE-5 card). Clicking on the test icons for Left and Right channels gives no sound but the animation plays as if the sounds are being played.

The `pavucontrol` detects both the nVidia HDMI card I don't use and the main SB AE-5 card, as : "HDA Creative Analog Stereo" with Port: "Line out (Plugged in)".

On the surface everything appears to be fine, but it isn't.

Audacious does not work with either pulseaudio or alsa. And normally it does work with both.

Both brave and firefox have no sound, and normally they both do. Video plays, no errors, gnome sound bar and pavucontrol both say it's all fine and the sound/volume/level meter bars are moving with the sound being pushed to the device but nothing ever comes out of the speakers.

Removing bundles 
`desktop, desktop-autostart and pulseaudio`
and adding bundles `sddm and desktop-awesome`
does not help - I can still run audacious with the ALSA output plugin but there is no sound.
Browser gives no sound, both firefox and brave.

I have no weird or exotic hardware, everything is high-end and just fine.

`aplay -L` gives:

> surround21
    2.1 Surround output to Front and Subwoofer speakers
surround40
    4.0 Surround output to Front and Rear speakers
surround41
    4.1 Surround output to Front, Rear and Subwoofer speakers
surround50
    5.0 Surround output to Front, Center and Rear speakers
surround51
    5.1 Surround output to Front, Center, Rear and Subwoofer speakers
surround71
    7.1 Surround output to Front, Center, Side, Rear and Woofer speakers
null
    Discard all samples (playback) or generate zero samples (capture)
hdmi:CARD=NVidia,DEV=0
    HDA NVidia, HDMI 0
    HDMI Audio Output
hdmi:CARD=NVidia,DEV=1
    HDA NVidia, HDMI 1
    HDMI Audio Output
hdmi:CARD=NVidia,DEV=2
    HDA NVidia, HDMI 2
    HDMI Audio Output
hdmi:CARD=NVidia,DEV=3
    HDA NVidia, HDMI 3
    HDMI Audio Output
hdmi:CARD=NVidia,DEV=4
    HDA NVidia, HDMI 4
    HDMI Audio Output
hdmi:CARD=NVidia,DEV=5
    HDA NVidia, HDMI 5
    HDMI Audio Output
hdmi:CARD=NVidia,DEV=6
    HDA NVidia, HDMI 6
    HDMI Audio Output
sysdefault:CARD=Creative
    HDA Creative, CA0132 Analog
    Default Audio Device
front:CARD=Creative,DEV=0
    HDA Creative, CA0132 Analog
    Front speakers
surround21:CARD=Creative,DEV=0
    HDA Creative, CA0132 Analog
    2.1 Surround output to Front and Subwoofer speakers
surround40:CARD=Creative,DEV=0
    HDA Creative, CA0132 Analog
    4.0 Surround output to Front and Rear speakers
surround41:CARD=Creative,DEV=0
    HDA Creative, CA0132 Analog
    4.1 Surround output to Front, Rear and Subwoofer speakers
surround50:CARD=Creative,DEV=0
    HDA Creative, CA0132 Analog
    5.0 Surround output to Front, Center and Rear speakers
surround51:CARD=Creative,DEV=0
    HDA Creative, CA0132 Analog
    5.1 Surround output to Front, Center, Rear and Subwoofer speakers
surround71:CARD=Creative,DEV=0
    HDA Creative, CA0132 Analog
    7.1 Surround output to Front, Center, Side, Rear and Woofer speakers
iec958:CARD=Creative,DEV=0
    HDA Creative, CA0132 Digital
    IEC958 (S/PDIF) Digital Audio Output

> 

I am attaching the alsa-info.sh output as a file (it is huge): 
[alsa-info.txt](https://github.com/clearlinux/distribution/files/4494658/alsa-info.txt)

`lsmod|grep snd`

> snd_hda_codec_ca0132    77824  1
snd_hda_codec_hdmi     61440  1
snd_hda_intel          53248  10
snd_intel_dspcfg       24576  1 snd_hda_intel
snd_hda_codec         135168  3 snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec_ca0132
snd_hda_core           90112  4 snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec,snd_hda_codec_ca0132
snd_hwdep              16384  1 snd_hda_codec
snd_pcm               118784  5 snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec,snd_hda_core,snd_hda_codec_ca0132
snd_timer              40960  1 snd_pcm
snd                    81920  27 snd_hda_codec_hdmi,snd_hwdep,snd_hda_intel,snd_hda_codec,snd_timer,snd_pcm,snd_hda_codec_ca0132
soundcore              16384  1 snd`
> 

the `lsmod|grep snd` is EXACTLY the same in opensuse tumbleweed. But in opensuse the gnome settings sound actually writes the name of the cardin the selector dropdown list for active device - Sound BlasterX, and there was soemthing else different, like "Core3D" or something similar (if I remember correctly). I always use 2 speakers stereo configuration in every distro and the speakers are always plugged right, although my card can do surround.

and some dmesg messages (but no errors):

`sudo dmesg|grep snd` (seems to be fine)
[dmesg_snd.txt](https://github.com/clearlinux/distribution/files/4494684/dmesg_snd.txt)

`sudo dmesg|grep sound` (seems to be fine)
[dmesg_sound.txt](https://github.com/clearlinux/distribution/files/4494739/dmesg_sound.txt)

`sudo dmesg|grep -i creative` (seems to be fine)
[dmesg_creative.txt](https://github.com/clearlinux/distribution/files/4494742/dmesg_creative.txt)

`journalctl` does give plenty of errors, however:
the search string I use is ALSA
these may not be all errors, including with the word ALSA
they are sequential, yet not neccessarily consecutive messages

> Apr 17 17:36:44 samara systemd-udevd[486]: controlC0: Process '/usr/bin/alsactl restore 0' failed with exit code 99.
...
Apr 17 22:47:47 samara pulseaudio[1206]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-alsa-card.so.avx2: module-alsa-card.so.avx2: cannot open shared object file: No such file or directory
...
Apr 17 22:47:47 samara pulseaudio[1206]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-alsa-card.so.avx2: module-alsa-card.so.avx2: cannot open shared object file: No such file or directory
...
Apr 17 19:55:01 samara pulseaudio[1201]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-alsa-card.so.avx2: module-alsa-card.so.avx2: cannot open shared object file: No such file or directory
...
Apr 17 19:55:01 samara pulseaudio[1201]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-alsa-card.so.avx2: module-alsa-card.so.avx2: cannot open shared object file: No such file or directory
...
...
...
Apr 17 21:36:55 samara pulseaudio[969]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-cli-protocol-unix.so.avx2: module-cli-protocol-unix.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara systemd[885]: pulseaudio.service: Succeeded.
Apr 17 21:36:58 samara systemd[885]: Starting Sound Service...
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-device-restore.so.avx2: module-device-restore.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-stream-restore.so.avx2: module-stream-restore.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-card-restore.so.avx2: module-card-restore.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-augment-properties.so.avx2: module-augment-properties.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-switch-on-port-available.so.avx2: module-switch-on-port-available.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-udev-detect.so.avx2: module-udev-detect.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-alsa-card.so.avx2: module-alsa-card.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-alsa-card.so.avx2: module-alsa-card.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-bluetooth-policy.so.avx2: module-bluetooth-policy.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-bluetooth-discover.so.avx2: module-bluetooth-discover.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-bluez5-discover.so.avx2: module-bluez5-discover.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-esound-protocol-unix.so.avx2: module-esound-protocol-unix.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-native-protocol-unix.so.avx2: module-native-protocol-unix.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-gsettings.so.avx2: module-gsettings.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-default-device-restore.so.avx2: module-default-device-restore.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-rescue-streams.so.avx2: module-rescue-streams.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-always-sink.so.avx2: module-always-sink.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-intended-roles.so.avx2: module-intended-roles.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-suspend-on-idle.so.avx2: module-suspend-on-idle.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-console-kit.so.avx2: module-console-kit.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-systemd-login.so.avx2: module-systemd-login.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-position-event-sounds.so.avx2: module-position-event-sounds.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-role-cork.so.avx2: module-role-cork.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-filter-heuristics.so.avx2: module-filter-heuristics.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara pulseaudio[6254]: E: [pulseaudio] ltdl-bind-now.c: Failed to open module module-filter-apply.so.avx2: module-filter-apply.so.avx2: cannot open shared object file: No such file or directory
Apr 17 21:36:58 samara systemd[885]: Started Sound Service.
>

`/usr/lib64/pulse-12.2/modules` is full of files, and they have the names of the ones not found according to journalctl, but the extensions are `.so`, not `.so.avx2`.

`pulseaudio --dump-modules` gives 
[pa-dm.txt](https://github.com/clearlinux/distribution/files/4494805/pa-dm.txt)

Sometimes audacious gives me the error:
> ALSA error: snd_pcm_open failed: Device or resource busy.
> 
I need a really really really fast linux, and I don't want to try any other - so, please can you fix this absurd bug ? I don't want to use another distro, I need clear linux.

There are two simmilar topics here and they were both closed, as a matter fo fact the authors claimed that everything is fine now. Well, I am getting that bug with every version of clear linux that I try (besides a very old one from last year). Google helps not. I tried a lot of topics ... and ... not a thing.

Thanx in advance,
Butterfly Collector

**Comment by ahkok** [comment]
Most important debug info I need: `sudo dmesg | grep firmware`.

**Comment by ahkok** [comment]
> The pavucontrol detects both the nVidia HDMI card I don't use

did you switch this to `off` in `pavucontrol`?

**Comment by butterflycollector** [comment]
`sudo dmesg|grep firmware`

> [    0.371487] calling  firmware_class_init+0x0/0x106 @ 1
[    0.371490] initcall firmware_class_init+0x0/0x106 returned 0 after 2 usecs
[    0.484789] calling  firmware_memmap_init+0x0/0x35 @ 1
[    0.484804] initcall firmware_memmap_init+0x0/0x35 returned 0 after 14 usecs
> 


In `pavucontrol` I cannot find an `off` switch, but the blue colored check for "set as fallback" is on the "HDA Creative analog stereo" and the little bar below the "Port:" selector is filling right and left when I try to play audio, as if the sound is sent to the Creative card and it's level/intensity is changing with the music. There is just no sound out of the speakers.

Plus my nVidia HDMI is connected to the monitor's speakers and there is nothing coming out of them when I play a sound. Unless I switch to the nVidia HDMI from the gnome > settings > sound panel and then the sound does come out of the monitor (but it sounds terrible ... bad integrated speakers, and the nVidia device is never known for good sound quality). I need the Creative, that is the real card - the rest is coincidental bonus from the hardware.

The integrated audio card from the motherboard is always off in the BIOS, so that one never appears.

**Comment by ahkok** [comment]
I didn't realize you have a separate audio card - do the other Linux OS's have a specific kernel module that they load? Can you check (e.g. ubuntu live CD?) and see if you can identify the module? We likely just don't have the right module enabled in our kernel.

**Comment by bwarden** [comment]
Do you have the module `snd-hda-codec-ca0132` loaded? This looks like the one, but also we should enable `CONFIG_SND_HDA_CODEC_CA0132_DSP` to get it to load the DSP firmware, `ctefx.bin`, which is provided in separate locations by bundles `alsa-utils` and `linux-firmware`.

**Comment by butterflycollector** [comment]
I guess I wrote too much stuff and it kind of gets lost ... I just wanted to give as much info, because I know it's meant to be helpful.

As I mentioned before there was a version of clear linux from september or october 2019 (the first time I tried it) which had sound and everything was perfect. I don't see a reason why this would change.

I also wrote the output of `lsmod | grep snd` (if this is good enough) and mentioned that it is identical to that of the current openSUSE tumbleweed. Literally every module, not one module more and not one module less and the sound is a problem only in clear linux.

I was just wondering if this was the module ... yes, snd_hda_codec_ca0132 is loaded.

OK, so how do I do that ... to enable `CONFIG_SND_HDA_CODEC_CA0132_DSP` ... and load `ctefx.bin` ?

**Comment by bwarden** [comment]
Could have been changes in the driver since then, not sure without digging deep into history. I'm enabling that option in our kernel now, which simply enables that behavior in the existing snd_hda_codec_ca0132 driver.

**Comment by butterflycollector** [comment]
Yes, I found two `ctefx.bin` : `/usr/lib/ctefx.bin` and `/usr/lib/firmware/ctefx.bin`. Are they going to load automatically ?

So, how do I update the kernel, just use swupd and add the package ? How do I check if it is already the right one (with `CONFIG_SND_HDA_CODEC_CA0132_DSP` enabled) ... I havent done this before ...

**Comment by bwarden** [comment]
> Oh, I see ... you mean linux-5.6.5-940 - so when it's out - I take it and hope.

Right. It has to go into the next release, which will probably happen on Monday. With this option enabled, the driver should automatically load the firmware. Unfortunately, my knowledge of the sound subsystem is pretty outdated at this point, so it'll take us more digging to figure out if there's something else that needs to happen, too.

**Comment by butterflycollector** [comment]
I see ... well, thanx to both of you for the effort and ... we'll see on Monday. :D

**Comment by ahkok** [comment]
Thanks for the very detailed report - that really helped.

**Comment by butterflycollector** [comment]
Hi again,

I actually tried and recompiled a kernel by following the clear linux docs on the internet and succeeded. In the `.config` I uncommented the option and added  `=y`:
`CONFIG_SND_HDA_CODEC_CA0132_DSP=y`. I don't know if it would work with `=m` or not. .... just in case this is useful. :D

Now I have sound.

There is another problem (I have not much experience with compiling kernels) - the nvidia driver cannot find the correct kernel headers ... at least one is missing in the source dir even after build (idk why), but I am not trying to open another topic. So, basically its either sound or 3d acceleration at the moment. Except after many tries of different things I simply kept the name of the kernel as in the repo and now I'm using the 937 native headers from the swupd bundle for the original kernel with the new compiled 937 native kernel.

But I sure hope everything will be fine on Monday, so I can have both sound and 3D acceleration the right way.

I hope all the right optimization flags for my temporary kernel have been applied automatically (I'm not sure I know how to set them since the last time I was trying to do something like that).

**Comment by bwarden** [comment]
> There is another problem (I have not much experience with compiling kernels) - the nvidia driver cannot find the correct kernel headers

Follow this guide to install NVIDIA drivers with dkms, which will automatically rebuild them when the new kernel to fix your sound issue is released:
https://docs.01.org/clearlinux/latest/tutorials/nvidia.html

**Comment by butterflycollector** [comment]
I did follow the tutorial and if I don't use the compiled kernel it works just fine, but if I use the compiled one it doesn't because it want some kind of headers. `make headers_install` from the kernel source dir is supposed to do the trick, but I haven't tried it yet. I decided to simply wait for the official release of the kernel.

Plus if I don't use the clear linux tools for deploying the kernel and kernel headers, I wonder can I easily switch between the compiled kernel headers and the bundled ones, hence can I use the boot manager option to choose different kernel - I would guess - not.

My major issue here is that the kernel `make build` process does not seem to be reading the `CFLAGS` and `CXXFLAGS` from the environment variables. ... so I just wait  for the release, where everything is built fine.
